docs/guide/snaps-rpc-api.md

```diff
@@ -382,71 +382,178 @@ const result = await ethereum.request({
 console.log(result); // We happen to know that this will be `true` or `false`
 ```
 
-### `snap_confirm`
+### `snap_dialog`
 
 ::: warning Only Callable By
 
 - Snaps
   :::
 
-#### Parameters
+#### Description
+
+Calling this method causes a dialog to be displayed in the MetaMask UI.
+There are three types of dialogs: Alert, Confirmation, and Prompt.
+Each of these dialog types has different parameters and return types, detailed below.
+
+##### Alert Dialog
+
+Displays an alert that can only be acknowledged.
+
+###### Parameters
 
 ```typescript
-interface SnapConfirmParam {
+interface SnapAlertDialogParam {
   /**
-   * A prompt, phrased as a question, no greater than 40 characters long.
+   * The alert title, no greater than 40 characters long.
    */
-  prompt: string;
+  title: string;
 
   /**
-   * A description, displayed with the prompt, no greater than 140 characters
-   * long.
+   * A description, displayed with the title, no greater
+   * than 140 characters long.
    */
   description?: string;
 
   /**
-   * Free-from text content, no greater than 1800 characters long.
-   * It will be displayed in monospace font in a scrollable text area.
+   * Free-from text content, no greater than 1800
+   * characters long.
    */
   textAreaContent?: string;
 }
 ```
 
 - `Array`
 
-  0. `SnapConfirmParam` - An object containing the contents of the confirmation.
+  0. `SnapAlertDialogParam` - An object containing the contents of the alert dialog.
 
-#### Returns
+###### Returns
 
-`boolean` - `true` if the user accepted the confirmation, and `false` otherwise.
+`null`
 
-#### Description
+###### Example
+
+```javascript
+await wallet.request({
+  method: 'snap_dialog',
+  params: [
+    {
+      type: DialogType.Alert,
+      title: 'Watch out!',
+      description: 'Look, a bird just flew by!',
+      textAreaContent: 'Birds are a group of warm-blooded vertebrates that are characterized by their ability to fly.',
+    },
+  ],
+});
```

Comment:
This is using a TS type therefore should be a TypeScript snippet.

`params` is not an array anymore.



```suggestion
```typescript
await wallet.request({
  method: 'snap_dialog',
  params: {
      type: DialogType.Alert,
      fields: {
      	title: 'Watch out!',
      	description: 'Look, a bird just flew by!',
      	textAreaContent: 'Birds are a group of warm-blooded vertebrates that are characterized by their ability to fly.',
      }
    },
});
```